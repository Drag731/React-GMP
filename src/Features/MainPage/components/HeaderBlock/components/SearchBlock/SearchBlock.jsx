import React from 'react';

// import './SearchBlock.scss'

const SearchBlock = ({handleSearch, search}) => (
    <div className="b-header__search">
        <span className="b-header__search-title">find your movie</span>
        <input
            onChange={(e) => handleSearch(e)}
            className="b-header__search-input"
            type="search"
            placeholder="search"
            value={search}
        />
    </div>
);

export default SearchBlock;
