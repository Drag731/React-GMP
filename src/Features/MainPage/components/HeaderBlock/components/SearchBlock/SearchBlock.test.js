import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import SearchBlock from './SearchBlock.jsx'

configure({ adapter: new Adapter() });

describe('<SearchBlock  />', () => {
    it('Snapshot one', () => {
        const handleSearch = jest.fn();
        const search = 'search';
        const tree = renderer
            .create(<SearchBlock
                handleSearch={handleSearch}
                search={search}
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Renders SearchBlock component', () => {
        const handleSearch = jest.fn();
        const search = 'search';
        const tree = shallow(
            <SearchBlock
                handleSearch={handleSearch}
                search={search}
            />
        );
        expect(tree.find('.b-header__search')).toHaveLength(1);
    });
    it('check SearchBlock component click', () => {
        const handleSearch = jest.fn();
        const search = 'search';
        const tree = shallow(
            <SearchBlock
                handleSearch={handleSearch}
                search={search}
            />
        );
        tree.find('input').simulate('change');

        expect(handleSearch).toHaveBeenCalled();
    });
});
