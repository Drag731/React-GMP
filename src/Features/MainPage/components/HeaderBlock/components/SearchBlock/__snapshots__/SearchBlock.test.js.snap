// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBlock  /> Snapshot one 1`] = `
<div
  className="b-header__search"
>
  <span
    className="b-header__search-title"
  >
    find your movie
  </span>
  <input
    className="b-header__search-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="search"
    type="search"
    value="search"
  />
</div>
`;
